hypertrace-gradle-java-convention-plugin/src/main/java/org/hypertrace/gradle/java/convention/JavaConventionPlugin.java

```diff
@@ -46,6 +47,15 @@ private void configureCompilerRelease(Project target) {
   }
 
   private void configureToolchain(Project target) {
+    if (javaConventionExtension(target)
+        .toolchainVersion
+        .map(v -> v.canCompileOrRun(JavaVersion.current().ordinal() + 1))
```

Comment:
The way this check is written is confusing for a few different reasons:
- the ordinal. They're not always meaningful (this one happens to be) and requires the +1. I'm pretty sure they expose a major version directly that could be used. 
- I think you can probably do the comparison as `JavaVersion`s which have more compatibility (you can do `JavaVersion.toVersion(JavaLanguageVersion)`).
- if the versions are the same, won't this fail? The same version on both sides for `canCompileAndRun` would return true and log